scripts/parse_lake_manifest_changes.py

```diff
@@ -0,0 +1,129 @@
+                    'url': new_pkg['url']
+                })
+        else:
+            changes.append({
+                'type': 'add',
+                'dependency': name,
+                'version': new_pkg['rev'],
+                'url': new_pkg['url']
+            })
+
+    # Find removed packages
+    for name in old_packages:
+        if name not in new_packages:
+            changes.append({
+                'type': 'remove',
+                'dependency': name,
+                'version': old_packages[name]['rev'],
+                'url': old_packages[name]['url']
+            })
+
+    return changes
+
+def get_github_diff_url(pkg_url, old_rev, new_rev):
+    """Convert a package URL to a GitHub compare URL"""
+    # For updates: compare URL
+    if old_rev and new_rev:
+        return f"{pkg_url}/compare/{old_rev}...{new_rev}"
+    # For additions: single new commit URL
+    elif new_rev is not None and old_rev is None:
+        return f"{pkg_url}/commit/{new_rev}"
+    # For removals: single old commit URL
+    elif old_rev is not None and new_rev is None:
+        return f"{pkg_url}/commit/{old_rev}"
+    # Handle case where no revisions are provided
+    else:
+        return pkg_url
+
+def format_changes(changes):
+    """Format the changes into a readable message"""
+    if not changes:
+        return "No dependency versions were changed"
+
+    def short_hash(hash):
+        """Return first 7 characters of commit hash"""
+        return hash[:7]
+
+    lines = []
+    updates = [c for c in changes if c['type'] == 'update']
+    additions = [c for c in changes if c['type'] == 'add']
+    removals = [c for c in changes if c['type'] == 'remove']
+
+    if updates:
+        lines.append("The following dependencies were updated:")
+        for change in updates:
+            url = get_github_diff_url(change['url'], change['old_version'], change['new_version'])
+            lines.append(f"* {change['dependency']}: {short_hash(change['old_version'])} → {short_hash(change['new_version'])} [[GitHub link]]({url})")
+
+    if additions:
+        if lines: lines.append("")
+        lines.append("The following dependencies were added:")
+        for change in additions:
+            url = get_github_diff_url(change['url'], None, change['version'])
+            lines.append(f"* {change['dependency']}: {short_hash(change['version'])} [[GitHub link]]({url})")
+
+    if removals:
+        if lines: lines.append("")
+        lines.append("The following dependencies were removed:")
+        for change in removals:
+            url = get_github_diff_url(change['url'], change['version'], None)
+            lines.append(f"* {change['dependency']}: {short_hash(change['version'])} [[GitHub link]]({url})")
+
+    return "\n".join(lines)
+
+def main():
+    old_manifest = get_json_at_rev('HEAD~1', 'lake-manifest.json')
+    if not old_manifest:
+        print("Failed to read old lake-manifest.json at HEAD~1")
+        return 1
```

Comment:
Could also use
```suggestion
        raise SystemExit("Failed to read old lake-manifest.json at HEAD~1")
```
and below (and then drop the `return 0` and `sys.exit()`)

Reply:
Done